int8 now_target
int8 grab_times
